messaging/src/main/java/org/axonframework/commandhandling/gateway/CommandDispatcher.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.commandhandling.gateway;
+
+import jakarta.annotation.Nonnull;
+import org.axonframework.common.infra.DescribableComponent;
+import org.axonframework.configuration.Configuration;
+import org.axonframework.messaging.Context;
+import org.axonframework.messaging.MetaData;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Component that publishes events to a {@link CommandGateway} in a predefined
+ * {@link org.axonframework.messaging.unitofwork.ProcessingContext context}. The events will be published in the context
+ * this appender was created for. You can construct one through the
+ * {@link #forContext(ProcessingContext, Configuration)}.
+ * <p>
+ * When using annotation-based {@link org.axonframework.messaging.annotation.MessageHandler @MessageHandler-methods} and
+ * you have declared an argument of type {@link CommandDispatcher}, the dispatcher will automatically be injected by the
+ * {@link org.axonframework.commandhandling.annotation.CommandDispatcherParameterResolverFactory}.
+ * <p>
+ * As this component is {@link ProcessingContext}-scoped, it is not retrievable from the {@link Configuration}.
+ *
+ * @author Steven van Beelen
+ * @since 5.0.0
+ */
+public interface CommandDispatcher extends DescribableComponent {
```

Comment:
So, bascially we have 3 levels of layering here:
- CommandBus
- CommandGateway
- CommandDispatcher

Using CommandDispatcher over CommandGateway is not very useful, if we do declarative components, for annotated - I see a lot of benefits. But I think we should carefully consider the naming. For now we have Event**Appender** / Command**Dispatcher**, but before in AF4 I known, as a user of the framework, generally what I was looking for (the API level I should use) was "Gateway" - same name with every concept.

Reply:
Alright, thanks for circling back! As a reaction to this comment, I did add a line in the JavaDoc of the `CommandDispatcher` (and `EventAppender`) clarifying the component is intended to be used within message handling functions.
Furthermore, I added a section about the `CommandDispatcher` to the (newly merged) Command Dispatching section of the `api-changes.md` to further clarify.